sdk/storage/storage-blob/src/generateSASUrl.ts

```diff
@@ -0,0 +1,22 @@
+import { BlobClient, GenerateSASUrlOptions } from "./BlobClient";
+import { SharedKeyCredential } from "./credentials/SharedKeyCredential";
+import { isNode } from "@azure/core-http";
+import { generateBlobSASQueryParameters } from "./BlobSASSignatureValues";
+
+BlobClient.prototype.generateSASUrl = function(
+  options: GenerateSASUrlOptions,
+  sharedKeyCredential: SharedKeyCredential
+): string {
+  if (isNode) {
+    return `${this.url}?${generateBlobSASQueryParameters(
```

Comment:
`this.url` might already contain a SAS token.

Reply:
Yes, Fixing.